### PR TITLE
recursive reveal

### PR DESCRIPTION
**Pull request by MenkeTechnologies**

### CODE REVIEW

File: reveal.sh [hunk 1]
```diff
@@ -9,20 +9,30 @@ reveal() {
...
-    $open_cmd https://github.com/$name?tab=repositories
-  fi
+    if [[ -z "$1" ]]; then
+        if ! command git rev-parse --git-dir 2> /dev/null 1>&2;then
+            "$open_cmd" "https://github.com/$name?tab=repositories"
+            return 0
+        fi
+    else
+        if ! command git rev-parse --git-dir 2> /dev/null 1>&2;then
+            for dir in "$@" ; do
```
Comment (MichaelDimmitt) [inline reveal.sh]: Looks like you are passing folder name in as an argument?

Why not open all sub folders using `for dir in "*" ; do`
Also its not "recursive reveal" its more like "subfolders reveal" since it only goes one level down.

File: reveal.sh [hunk 2]
```diff
@@ -9,20 +9,30 @@ reveal() {
...
+                ( builtin cd "$dir" && reveal; )
+            done
+            return 0
+        fi
```
Comment (MichaelDimmitt) [inline reveal.sh]: Looks like you are passing folder name in as an argument?
```bash
for dir in "$@" ; do
  ( builtin cd "$dir" && reveal; )
done
```
Why not open all sub folders using 
```bash
for d in */; do
  ( builtin cd "$dir" && reveal; )
done
```
Also its not "recursive reveal" its more like "subfolders reveal" since it only goes one level down. 

It might be cool to pass in arguments. Curious see your thoughts.
Of course you are a regular user so your workflow is important.

**Comment by MichaelDimmitt** [comment]
See #27 for discussion about improving the readability of the pull request.
thanks.

**Comment by MenkeTechnologies** [comment]
1. previous behavior if no remote specified
... opens all remotes available for repo.
2. previous behavior if remote specified as an argument such as 'origin'
... opens only the origin remote.

Both of these behaviors have been maintained.  The only difference Is the behavior when not in a git project.  In this case with no args it opens up the users repositories.  If arguments are given then cd to each arg and reveal all the remotes there.  No filter on the name of the remote in that case.  This was the behavior I wanted so I could do reveal * or reveal **/*.  The recursive was in reference to the code that calls reveal again.

```
for d in */; do
  ( builtin cd "$dir" && reveal; )
done
```
is too wordy for my tired fingers
